src/util/check.h

```diff
@@ -42,4 +42,12 @@ class NonFatalCheckError : public std::runtime_error
         }                                                         \
     } while (false)
 
+/** Assert that the passed pointer is not nullptr and return a reference to the pointed object */
+template <typename PtrT>
+typename std::remove_reference<decltype(*PtrT{})>::type& Ensure(const PtrT& ptr)
```

Comment:
In commit "wallet: Pass unused args to WalletClientImpl" (fa38755d2fc444589ae506e4edd8597ab25e55e9)

Slight preference for calling this `AssertNotNull` instead of `Ensure` just because it could wind up being used widely, and it'd be nice for it to be obvious what it does.

Also for code review purposes, C++ programmers are already pretty trained to know that `*ptr` is dangerous and can kill the program, while `Ensure(ptr)` might look more innocuous if it's part of a larger expression. And since the name doesn't contain "Assert" or "Abort", it might actually be clearer if this returned a pointer so the caller is forced to write `*`. Could also go more generic and provide

```c++
template<typename T> T&& Assert(T&& value) { assert(value); return std::forward<T>(value); }
```